pkg/mcs/resourcemanager/server/manager.go

```diff
@@ -116,6 +119,25 @@ func (m *Manager) GetStorage() endpoint.ResourceGroupStorage {
 	return m.storage
 }
 
+// GetKeyspaceServiceLimiter returns the service limit of the keyspace.
+func (m *Manager) GetKeyspaceServiceLimiter(keyspaceID uint32) *serviceLimiter {
+	krgm := m.getKeyspaceResourceGroupManager(keyspaceID)
+	if krgm == nil {
+		return nil
+	}
+	return krgm.getServiceLimiter().Clone()
+}
+
+// SetKeyspaceServiceLimit sets the service limit of the keyspace.
+func (m *Manager) SetKeyspaceServiceLimit(keyspaceID uint32, serviceLimit float64) {
+	// If the keyspace is not found, create a new keyspace resource group manager.
+	krgm := m.getOrCreateKeyspaceResourceGroupManager(keyspaceID)
+	if krgm == nil {
```

Comment:
Seems this is an unreachable path as `getOrCreateKeyspaceResourceGroupManager` should never fail. If it can actually retrun `nil` value, I think you should return an error as `SetKeyspaceServiceLimit` is not successful here.